test/unit/test_matrix.cpp

```diff
@@ -0,0 +1,365 @@
+    {{{31, 17}, {7, 10}, 31, 7, 341},     // Scalapack like layout
+     {{31, 17}, {7, 11}, 33, 7, 363},     // with padding (ld)
+     {{31, 17}, {7, 11}, 47, 11, 517},    // with padding (row)
+     {{31, 17}, {7, 11}, 31, 7, 348},     // with padding (col)
+     {{29, 41}, {13, 11}, 13, 143, 429},  // Tile like layout
+     {{29, 41}, {13, 11}, 17, 183, 549},  // with padding (ld)
+     {{29, 41}, {13, 11}, 13, 146, 438},  // with padding (row)
+     {{29, 41}, {13, 11}, 13, 143, 436},  // with padding (col)
+     {{29, 41}, {13, 11}, 13, 143, 419},  // compressed col_offset
+     {{0, 0}, {1, 1}, 1, 1, 1}});
+
+TYPED_TEST(MatrixTest, ConstructorExisting) {
+  using Type = TypeParam;
+
+  for (const auto& v : values) {
+    auto size = std::get<0>(v);
+    auto block_size = std::get<1>(v);
+    auto ld = std::get<2>(v);
+    auto row_offset = std::get<3>(v);
+    auto col_offset = std::get<4>(v);
+
+    matrix::LayoutInfo layout(size, block_size, ld, row_offset, col_offset);
+    memory::MemoryView<Type, Device::CPU> mem(layout.minMemSize());
+
+    Matrix<Type, Device::CPU> mat(layout, mem(), mem.size());
+
+    CHECK_FROM_EXISTING(mem(), layout, mat);
+  }
+}
+
+TYPED_TEST(MatrixTest, ConstructorExistingConst) {
+  using Type = TypeParam;
+
+  for (const auto& v : values) {
+    auto size = std::get<0>(v);
+    auto block_size = std::get<1>(v);
+    auto ld = std::get<2>(v);
+    auto row_offset = std::get<3>(v);
+    auto col_offset = std::get<4>(v);
+
+    matrix::LayoutInfo layout(size, block_size, ld, row_offset, col_offset);
+    memory::MemoryView<Type, Device::CPU> mem(layout.minMemSize());
+
+    const Type* p = mem();
+    Matrix<const Type, Device::CPU> mat(layout, p, mem.size());
+
+    CHECK_FROM_EXISTING(mem(), layout, mat);
+  }
+}
+
+/// @brief Returns true if the first @p ready futures are ready.
+/// @pre Future* should be future or shared_future
+/// @pre 0 <= ready <= futures.size()
+template <template <class> class Future, class T, Device device>
+bool checkFuturesStep(size_t ready, const std::vector<Future<Tile<T, device>>>& futures) {
+  assert(ready >= 0);
+  assert(ready <= futures.size());
+
+  for (std::size_t index = 0; index < ready; ++index) {
+    if (!futures[index].is_ready())
+      return false;
+  }
+  for (std::size_t index = ready; index < futures.size(); ++index) {
+    if (futures[index].is_ready())
+      return false;
+  }
+  return true;
+}
+
+/// @brief Checks if current[i] depends correctly on previous[i].
+/// If get_ready == true it checks if current[i] is ready after previous[i] is used.
+/// If get_ready == false it checks if current[i] is not ready after previous[i] is used.
+/// @pre Future* should be future or shared_future
+template <template <class> class Future1, template <class> class Future2, class T1, class T2,
+          Device device>
+void checkFutures(bool get_ready, const std::vector<Future1<Tile<T1, device>>>& current,
```

Comment:
Can we find a better name? In this function, you are not just checking `current`, but you are also triggering `previous`. I would make it more explicit.

IMHO it would make more clear also the role of `get_ready`, it could become something like `current_ready_after_trigger`.

Reply:
Ok that the name should explain what the function does, but here we are going in the direction that the function name should contain the full documentation.